app/src/main/java/org/acmvit/gitpositive/MainActivity.kt

```diff
@@ -1,80 +1,90 @@
 package org.acmvit.gitpositive
 
-import android.graphics.Color
-import android.graphics.drawable.ColorDrawable
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import android.text.Html
 import android.view.LayoutInflater
-import android.view.View
-import android.widget.TextView
 import android.widget.Toast
+import androidx.activity.viewModels
 import androidx.appcompat.app.AlertDialog
 import com.bumptech.glide.Glide
 import org.acmvit.gitpositive.databinding.ActivityMainBinding
-import retrofit2.Call
-import retrofit2.Callback
-import retrofit2.Response
-import retrofit2.Retrofit
-import retrofit2.converter.gson.GsonConverterFactory
 
-var BaseURL="https://api.github.com/"
+
 class MainActivity : AppCompatActivity() {
-    private var loadingDialog: AlertDialog? = null
+
+
+    private val viewModel by viewModels<MainViewModel>()
+    private var _binding: ActivityMainBinding? = null
+    private val binding: ActivityMainBinding
+        get() {
+            return _binding!!
+        }
```

Comment:
Pretty smart to use a getter, but I would highly suggest to not use doublebang operator! There is a scope of optimisation here.

Reply:
Hi @DarthBenro008 this one is backing field. 
Please find the reference here : [SO link](https://stackoverflow.com/questions/66442242/purpose-of-with-without-underscore-variable-view-binding)